zb-db/src/test/java/io/camunda/zeebe/db/impl/ColumnFamilyRandomizedPropertyTest.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.db.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.ZeebeDbFactory;
+import io.camunda.zeebe.db.ZeebeDbInconsistentException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import net.jqwik.api.Arbitraries;
+import net.jqwik.api.Combinators;
+import net.jqwik.api.ForAll;
+import net.jqwik.api.Property;
+import net.jqwik.api.Provide;
+import net.jqwik.api.arbitraries.ListArbitrary;
+import net.jqwik.api.lifecycle.AfterProperty;
+import net.jqwik.api.lifecycle.AfterTry;
+import net.jqwik.api.lifecycle.BeforeProperty;
+import org.assertj.core.util.Files;
+
+public class ColumnFamilyRandomizedPropertyTest {
+
+  private Map<Long, Long> map;
+  private ColumnFamily<DbLong, DbLong> columnFamily;
+  private ZeebeDb<DefaultColumnFamily> zeebeDb;
+
+  @BeforeProperty
+  private void setup() {
+    final var pathName = Files.newTemporaryFolder();
+    final ZeebeDbFactory<DefaultColumnFamily> dbFactory = DefaultZeebeDbFactory.getDefaultFactory();
+    zeebeDb = dbFactory.createDb(pathName);
+
+    columnFamily =
+        zeebeDb.createColumnFamily(
+            DefaultColumnFamily.DEFAULT, zeebeDb.createContext(), new DbLong(), new DbLong());
+    map = new HashMap<>();
+  }
+
+  @Property
+  void columnFamilyHasSameEntriesAsMapAfterModifying(
+      @ForAll("operations") final Iterable<TestableOperation> operations) {
```

Comment:
:wrench: more a nit pick: in my mind the testable operation and the key/value are independent domains.

So I would prefer a signature like 
```
columnFamilyHasSameEntriesAsMapAfterModifying(@ForAll @LongRange(0, Long.MAX_VALUE) long key, @ForAll @LongRange(Long.MIN_VALUE, Long.MAX_VALUE) long value, Iterable<Operateion> operations)
```

And then combine these independent operations in the test, rather than combining them in data generation. 
But, admittedly, this is a very subjective preference.

Reply:
Hm, but it'd like to test a sequence of operations and each operation needs to be associated with one kv-pair. So I'd have to generate a list of keys, a list of values and a list of operations (and all lists must have the same length). In your example, the test would run for one key, one value and a list of operations. That's not exactly what I want to test here. I'll leave it as is for now :)